[Refactor] Api

YongLeeCode opened the pull request:
<!-- PR 제목은 " [Feat] landing-page" 형식으로 작성 -->
<!-- PR open 후 링크를 복사해 노션에 추가 -->

## ✅ 체크 리스트

<!-- PR 승인을 받기전 놓친 것 없는지 확인해주세요. -->

- [ ] 적절한 Title 작성
- [ ] 적절한 Label 지정
- [ ] Assignee 및 Reviewer 지정
- [ ] 로컬 작동 확인
- [ ] Merge 되는 브랜치 확인

## 📝 작업 내용

<!-- 작업한 내용에 대해 작성해주세요. -->
주요 작업은 로그인과 회원가입 페이지 리펙토링 작업을 했습니다.
필요없는 부분, 읽기 쉬운 코드로 만들려고 노력했습니다.
그리고 instance를 두 종류로 나눴습니다.
instance와 instanceWithoutAccessToken으로요.
이름 그대로 accessToken이 필요한 것과 필요없는 인스턴스를 나눴는데, 이유는 리이슈 인터셉터 과정에서 무한 반복되는 문제가 있었기 때문입니다.

## 🖥️ 공유 포인트 및 논의 사항

<!-- 공유하거나 논의할 사항을 작성해주세요. -->
혹시 사용하다가 문제가 있거나 에러가 생기면 즉시 알려주세요.

YongLeeCode committed:
refactor: 이미지 코드 리펙토링

YongLeeCode committed:
feat: 유저 타입 정리

YongLeeCode committed:
refactor: 업데이트 패키지 버전

YongLeeCode committed:
refactor: 유저 api 분리

YongLeeCode committed:
refactor: 인스턴스 변화로 인한 코드 변경

YongLeeCode committed:
refactor: 인스턴스 변화로 인한 코드 변경

YongLeeCode committed:
refactor: 로그인폼 컴포넌트 생성

YongLeeCode committed:
refactor: 회원가입폼 컴포넌트 생성

han9898 reviewed (approved):
👍 로그인 페이지에서 계정 정보를 틀리게 적었을 시에 잘 동작하네요

inji0212 reviewed (approved):
수고하셨습니다!

YongLeeCode committed:
refactor: api 파일 리펙토링

YongLeeCode committed:
feat: 상태관리 리액트 쿼리 사용

- 원래 방법은 유저의 정보를 response로 받아서 zustand에 넣고 그것을 localStorage에 넣는 방식이였다.
- 이번에 바꾼 방법: response로 받은 것을 react query로 staletime과 gcTime을 이용해서 데이터의 신선한 시간을 조절했다.
- 이제 더 이상 localStorage에 안 넣어도 매번 데이터를 백엔드에서 불러오는 일이 줄어들었다.

edhcoding reviewed (approved):
고생하셨습니다!

YongLeeCode committed:
Merge branch 'dev' into page-login
